lib/bubbles.js

```diff
@@ -21,23 +21,40 @@ Bubble.prototype.draw = function(context) {
 };
 
 Bubble.prototype.move = function() {
-  if (this.status === "new") {
-    this.x+=this.xInc;
-  } else if (this.status === "floating") {
-    this.y--;
+  var bubble = this;
+  float(bubble);
+  bubble.count++;
+  if (doneFloatingSidewaysOrHitAWall(bubble)) {
+    bubble.status = "floating";
   }
-  this.count++;
-  if (this.x >= (this.canvas.width - 5 ) || this.x <= 0 || this.count === 50) {
-    this.status = "floating";
+  if (onRightEdge(bubble)) {
+    bubble.x = bubble.canvas.width - bubble.width;
   }
-  if (this.x >= this.canvas.width - this.width) {
-    this.x = this.canvas.width - this.width;
+  if (onCeiling(bubble)) {
+    bubble.status = "done";
   }
-  if (this.y <= 0) {
-    this.status = "done";
-  }
-
-  return this;
+  return bubble;
 };
 
+function doneFloatingSidewaysOrHitAWall(bubble) {
+  return bubble.x >= (bubble.canvas.width - bubble.width ) || bubble.x <= 0 || bubble.count === 50;
```

Comment:
It does, but the branches have different behavior - one sets the bubble's status, onRightEdge sets the bubble's x in case we overshot, so we'll want to keep them separate.